OpenRA.Mods.Common/Scripting/Properties/CaptureProperties.cs

```diff
@@ -20,24 +22,26 @@ namespace OpenRA.Mods.Common.Scripting
 	[ScriptPropertyGroup("Ability")]
 	public class CaptureProperties : ScriptActorProperties
 	{
-		readonly CapturesInfo normalInfo;
+		readonly Captures[] capturesTraits;
 		readonly ExternalCapturesInfo externalInfo;
 
 		public CaptureProperties(ScriptContext context, Actor self)
 			: base(context, self)
 		{
-			normalInfo = Self.Info.TraitInfoOrDefault<CapturesInfo>();
+			capturesTraits = Self.TraitsImplementing<Captures>().ToArray();
 			externalInfo = Self.Info.TraitInfoOrDefault<ExternalCapturesInfo>();
 		}
 
 		[Desc("Captures the target actor.")]
 		public void Capture(Actor target)
 		{
+			var captures = capturesTraits.FirstOrDefault(x => !x.IsTraitDisabled && x.Info.CaptureTypes.Overlaps(target.GetEnabledTargetTypes()));
```

Comment:
You're checking against Targetable->Types here and not Capturable->Types. This block of code should be rewritten IMO as
```
var normalCapturable = target.Info.TraitInfoOrDefault<CapturableInfo>();

if (normalCapturable != null)
{
	var captures = capturesTraits.FirstOrDefault(x => !x.IsTraitDisabled && x.Info.CaptureTypes.Contains(normalCapturable.Type));
	if (captures != null)
	{
		Self.QueueActivity(new CaptureActor(Self, target, captures));
		return;
	}
}

var externalCapturable = target.Info.TraitInfoOrDefault<ExternalCapturableInfo>();

if (externalInfo != null && externalCapturable != null && externalInfo.CaptureTypes.Contains(externalCapturable.Type))
	Self.QueueActivity(new ExternalCaptureActor(Self, Target.FromActor(target)));
else
```